### DISCUSSION TITLE
Can headers in files be kept?

### DISCUSSION THREAD

**Discussion by hansva** [WeblateOrg/weblate#11690] [Q&A] [answered]
Hi all,

First of all, thanks for this great product/project. I am part of an Apache Software Foundation project and we are actively using Weblate to translate our UI into other languages. 

I recently upgraded our Weblate instance to use system-wide add-ons, as we are a project with over 200 components and I did not want to update each component separately. The add-ons of interest are "Format the Java properties file" and "Cleanup translation files". Since the update however I have an unwanted side effect. Our files must contain the ASF header and Weblate is now actively removing these headers [Sample PR](https://github.com/apache/hop/pull/3958/files). I already disabled the "Format the Java properties file" plugin, but the issue seems to persist.

As I changed both the Weblate version and added plugins I am having difficulties determining the root cause. Unfortunately I do not remember exactly which version we upgraded from but it will still have been in the 4.X range. Any advice or solution would be greatly appreciated!

Kr,
Hans

**Comment by nijel** [comment, marked as answer]
Unfortunately, this is not supported in the translate-toolkit for the properties files.

**Reply by hansva** [reply]
Thank you for the reply, then it was accidental that this used to work.
Is there an easy way to run pre-commit scripts or would I have to write my own add-on to implement this?

**Reply by nijel** [reply]
I think if there is no space between the comment and a property, it will parse it as a property comment and those are kept. The file-level comments are not kept.

**Reply by hansva** [reply]
That used to be the case, at some point I added the newlines because adding the header as a comment to the property caused it to jump around in the file if keys were added that sorted above the first one. Thanks for the answers.
